### ISSUE TITLE
Add arbitrary layer overrides

### ISSUE THREAD

**Issue by beasteers** [rochacbruno/dynaconf#121]
It would be nice if we could add arbitrarily layered config options at runtime, especially when working with configs across modules. For example:
```yaml
default:
    option1: blah
    option2: blah
    option3: blah

development:
    option1: blahh
    option2: blahhh

    some_variation:
        option1: blahhhhhh

    another_variation:
        option1: bla
```

```python
from dynaconf import settings as cfg

print(cfg.OPTION1) # blahh

with cfg.useopts(**cfg.SOME_VARIATION):
        print(cfg.OPTION1) # blahhhhhh

print(cfg.OPTION1) # blahh
```
And analogous to the `setenv` API
```python
print(cfg.OPTION1) # blahh

cfg.setopts('some_layer', optionA='blah', **cfg.SOME_VARIATION)
print(cfg.OPTION1) # blahhhhhh
print(cfg.OPTIONA) # blah

cfg.setopts('another_layer', optionA='blahhhhhh')
print(cfg.OPTION1) # blahhhhhh
print(cfg.OPTIONA) # blahhhhhh

cfg.removeopts('another_layer')
cfg.setopts('some_layer', **cfg.ANOTHER_VARIATION) # replace previously defined 'some_layer'
print(cfg.OPTION1) # bla
print(cfg.OPTIONA) # raise

cfg.removeopts('some_layer')
print(cfg.OPTION1) # blahh
```
It seems like a convenient generalization of the file overrides.

**Comment by rochacbruno** [comment]
after #146 this can be a plugin

**Comment by rochacbruno** [comment]
This is fixed by #219 now you can use `settings.from_env`
